core/vtk/ttkCompactTriangulationPreconditioning/ttkCompactTriangulationPreconditioning.cpp

```diff
@@ -0,0 +1,203 @@
+    return 0;
+
+  // If all checks pass then log which array is going to be processed.
+  this->printMsg("Starting computation...");
+
+  ttk::Triangulation *triangulation
+    = ttkAlgorithm::GetTriangulation(inputDataSet);
+  if(!triangulation)
+    return 0;
+
+  // Precondition the triangulation (e.g., enable fetching of vertex neighbors)
+  // this->preconditionTriangulation(triangulation); // implemented in base
+  // class
+
+  // Templatize over the different input array data types and call the base code
+  int status = 0; // this integer checks if the base code returns an error
+  ttkTemplateMacro(
+    triangulation->getType(),
+    (status = this->execute((TTK_TT *)triangulation->getData(), Threshold)));
+
+  // On error cancel filter execution
+  if(status != 1)
+    return 0;
+
+  // Get output vtkDataSet (which was already instantiated based on the
+  // information provided by FillOutputPortInformation)
+  vtkPointSet *outputDataSet = vtkPointSet::GetData(outputVector, 0);
+  vector<SimplexId> vertexMap(this->vertices.size());
+  vtkUnstructuredGrid *outputMesh
+    = vtkUnstructuredGrid::SafeDownCast(outputDataSet);
+
+  vtkSmartPointer<vtkPoints> points = vtkSmartPointer<vtkPoints>::New();
+  vtkSmartPointer<vtkIntArray> indices = vtkSmartPointer<vtkIntArray>::New();
+
+  indices->SetNumberOfComponents(1);
+  indices->SetName(compactTriangulationIndex);
+
+  // insert the vertices in the output mesh
+  for(size_t i = 0; i < this->vertices.size(); i++) {
+    float x, y, z;
+    triangulation->getVertexPoint(this->vertices.at(i), x, y, z);
+    points->InsertNextPoint(x, y, z);
+    vertexMap[this->vertices.at(i)] = i;
+    indices->InsertNextTuple1(this->nodes.at(i));
+  }
+  outputMesh->SetPoints(points);
+
+  vtkPointData *pointData = outputMesh->GetPointData();
+  pointData->AddArray(indices);
+
+  std::vector<std::string>::iterator iter = scalarFields.begin();
+  while(iter != scalarFields.end()) {
+    vtkDataArray *inputArray
+      = inputDataSet->GetPointData()->GetArray(iter->data());
+    if(inputArray == nullptr) {
+      iter++;
+      continue;
+    }
+
+    iter = scalarFields.erase(iter);
+    vtkDataArray *newField = nullptr;
+
+    // // To make sure that the selected array can be processed by this filter,
+    // // one should also check that the array association and format is
+    // correct. if(this->GetInputArrayAssociation(0, inputVector) != 0) {
+    //   this->printErr("Input array needs to be a point data array.");
+    //   return 0;
+    // }
+
+    switch(inputArray->GetDataType()) {
+
+      case VTK_CHAR:
+        newField = vtkCharArray::New();
+        break;
+
+      case VTK_DOUBLE:
+        newField = vtkDoubleArray::New();
+        break;
+
+      case VTK_FLOAT:
+        newField = vtkFloatArray::New();
+        break;
+
+      case VTK_INT:
+        newField = vtkIntArray::New();
+        break;
+
+      case VTK_ID_TYPE:
+        newField = vtkIdTypeArray::New();
+        break;
+    }
```

Comment:
We might need a `default` label here (all possible VTK types are not covered). Or, in a simpler way, replace the whole switch/case with `vtkSmartPointer<vtkDataArray> newField{inputArray->NewInstance()};`?